code/controller/actuator.cpp

```diff
@@ -1,10 +1,176 @@
+#include <thread>
+#include <chrono>
+
 #include "actuator.h"
+#include "../utility/logger.h"
+
+Actuator::Actuator(const QString& serial_port, const PortSettings& settings, QextSerialPort::QueryMode mode) {
+    m_serial_port = new QextSerialPort(serial_port, settings, mode);
+	m_invert_x = 1;
+	m_invert_y = 1;
+    //on successful connection
+    if (m_serial_port->lastError() == 0) {
+        Logger::log(serial_port.toStdString() + " successfully opened!", Logger::INFO);
+        m_serial_port->flush();
+		// Send a renumbering request for the devices
+		setDeviceNumber();
+    }
+    else {
+        Logger::log("ERROR: " + serial_port.toStdString() + " could not be opened! " + m_serial_port->errorString().toStdString(), Logger::ERROR);
+    }
+}
+
+Actuator::Actuator(const Actuator& other) {
+	m_x_device = other.m_x_device;
+	m_y_device = other.m_y_device;
+	m_invert_x = other.m_invert_x;
+	m_invert_y = other.m_invert_y;
+
+	m_serial_port = other.m_serial_port;
+}
+
+char* const Actuator::convertDataToBytes(long int data) {
+	if (data < 0) {
+		data = intPow(BYTE_RANGE, 4) + data;
+	}
+
+	char result[DATA_SIZE];
+	
+	for (int i = DATA_SIZE - 1; i >= 0; --i) {
+		int temp = intPow(BYTE_RANGE, i);
+		result[i] = data / temp;
+		data = data - temp * result[i];
+	}
+
+	return result;
+}
+
+void Actuator::setDeviceNumber() {

```

Comment:
it is confusing how the Zaber API does this thing, it's a renumbering of the devices that has to be called when setting it up for the first time, so yeah `resetDeviceNumber` is probably more appropriate.